### PR TITLE
doc: Expand scantxoutset help text to cover tr() and miniscript

### PR DESCRIPTION
**Pull request by instagibbs**

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -2099,13 +2099,16 @@ static RPCHelpMan scantxoutset()
...
+        "\nScans the unspent transaction output set for entries that match certain output descriptors or miniscripts.\n"
         "Examples of output descriptors are:\n"
         "    addr(<address>)                      Outputs whose scriptPubKey corresponds to the specified address (does not include P2PK)\n"
         "    raw(<hex script>)                    Outputs whose scriptPubKey equals the specified hex scripts\n"
         "    combo(<pubkey>)                      P2PK, P2PKH, P2WPKH, and P2SH-P2WPKH outputs for the given pubkey\n"
         "    pkh(<pubkey>)                        P2PKH outputs for the given pubkey\n"
         "    sh(multi(<n>,<pubkey>,<pubkey>,...)) P2SH-multisig outputs for the given threshold and pubkeys\n"
+        "    tr(<pubkey>)                         P2TR\n"
+        "    tr(<pubkey>,{pk(<pubkey>)})          P2TR with single fallback pubkey in tapscript\n"
+        "    wsh(and_v(v:pk(<pubkey>),after(2)))  P2WSH miniscript with mandatory pubkey and a timelock\n"
```
Comment (w0xlt) [inline src/rpc/blockchain.cpp:2111]: Could also include `rawtr(<pubkey>) P2TR output with the specified key as output key`.

**Review by w0xlt — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/27155/commits/ca605f015dc8fbabbc6c0640d87429d6bf8f553f

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -2099,13 +2099,16 @@ static RPCHelpMan scantxoutset()
     const std::string EXAMPLE_DESCRIPTOR_RAW = "raw(76a91411b366edfc0a8b66feebae5c2e25a7b6a5d1cf3188ac)#fm24fxxy";
 
     return RPCHelpMan{"scantxoutset",
-        "\nScans the unspent transaction output set for entries that match certain output descriptors.\n"
+        "\nScans the unspent transaction output set for entries that match certain output descriptors or miniscripts.\n"
```
Comment (darosior) [inline src/rpc/blockchain.cpp:2102]: nit: i don't think it's worth mentioning Miniscript here, and especially not as something external to descriptors.

**Review by darosior — APPROVED** [review]
ACK ca605f015dc8fbabbc6c0640d87429d6bf8f553f -- thanks for improving the doc.

**Comment by Sjors** [comment]
Concept ACK. Note that the help already refers to `doc/descriptors.md` so it doesn't have to be super thorough.

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (instagibbs) [reply src/rpc/blockchain.cpp:2102]: hmm true, will remove

**Comment by instagibbs** [comment]
Wondering if the help text should just give hints to what possible types there are, and point users to the doc in question to map those types to their required formats?

Should we version the `doc/descriptors.md` file?

**Comment by fanquake** [comment]
> Should we version the doc/descriptors.md file? By that I mean literally add details in the doc saying when specific formats were added.

cc @achow101

**Comment by achow101** [comment]
ACK ca605f015dc8fbabbc6c0640d87429d6bf8f553f

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (achow101) [reply src/rpc/blockchain.cpp:2102]: @instagibbs Are you planning on changing this?

**Comment by achow101** [comment]
> Should we version the `doc/descriptors.md` file? By that I mean literally add details in the doc saying when specific formats were added.

Probably.

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (instagibbs) [reply src/rpc/blockchain.cpp:2112]: added a version of it, thanks

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (instagibbs) [reply src/rpc/blockchain.cpp]: yeah, pushed the change.

**Review by darosior — APPROVED** [review]
webACK e4ede64fe87b1619f65da5753088d93e6f481b05

**Pull request merged**